File: marketplace_app/lib/models/specialist.dart
class SpecialistService {
  final String id;
  final String serviceName;
  final String category;
  final int durationMinutes;
  final double price;
  final String? description;

  SpecialistService({
    required this.id,
    required this.serviceName,
    required this.category,
    required this.durationMinutes,
    required this.price,
    this.description,
  });

  factory SpecialistService.fromJson(Map<String, dynamic> json) {
    return SpecialistService(
      id: json['id'],
      serviceName: json['serviceName'] ?? '',
      category: json['category'] ?? '',
      durationMinutes: json['durationMinutes'] ?? 0,
      price: (json['price'] as num?)?.toDouble() ?? 0.0,
      description: json['description'],
    );
  }
}

class ServiceArea {
  final String city;
  final String? postalCode;
  final int maxDistanceKm;
  final bool isPrimary;

  ServiceArea({
    required this.city,
    this.postalCode,
    required this.maxDistanceKm,
    required this.isPrimary,
  });

  factory ServiceArea.fromJson(Map<String, dynamic> json) {
    return ServiceArea(
      city: json['city'] ?? '',
      postalCode: json['postalCode'],
      maxDistanceKm: json['maxDistanceKm'] ?? 0,
      isPrimary: json['isPrimary'] ?? false,
    );
  }
}

class Specialist {
  final String id;
  final String firstName;
  final String lastName;
  final String? professionalTitle;
  final String? bio;
  final double? hourlyRate;
  final bool isVerified;
  final double averageRating;
  final int totalReviews;
  final List<SpecialistService> services;
  final List<ServiceArea> serviceAreas;
  final double? distance;

  Specialist({
    required this.id,
    required this.firstName,
    required this.lastName,
    this.professionalTitle,
    this.bio,
    this.hourlyRate,
    required this.isVerified,
    required this.averageRating,
    required this.totalReviews,
    required this.services,
    required this.serviceAreas,
    this.distance,
  });

  String get fullName => '$firstName $lastName';

  factory Specialist.fromJson(Map<String, dynamic> json) {
    return Specialist(
      id: json['id'],
      firstName: json['firstName'] ?? '',
      lastName: json['lastName'] ?? '',
      professionalTitle: json['professionalTitle'],
      bio: json['bio'],
      hourlyRate: (json['hourlyRate'] as num?)?.toDouble(),
      isVerified: json['isVerified'] ?? false,
      averageRating: (json['averageRating'] as num?)?.toDouble() ?? 0.0,
      totalReviews: json['totalReviews'] ?? 0,
      services:
          (json['services'] as List?)
              ?.map((e) => SpecialistService.fromJson(e))
              .toList() ??
          [],
      serviceAreas:
          (json['serviceAreas'] as List?)
              ?.map((e) => ServiceArea.fromJson(e))
              .toList() ??
          [],
      distance: (json['distance'] as num?)?.toDouble(),
    );
  }
}

File: backend/H4H.Core/Models/Specialist.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace H4H.Core.Models
{
    [Table("specialists")]
    public class Specialist
    {
        [Column("id")]
        public Guid Id { get; set; }

        [Column("user_id")]
        public Guid UserId { get; set; }

        [Column("first_name")]
        public string FirstName { get; set; } = string.Empty;

        [Column("last_name")]
        public string LastName { get; set; } = string.Empty;

        [Column("professional_title")]
        public string? ProfessionalTitle { get; set; }

        [Column("bio")]
        public string? Bio { get; set; }

        [Column("hourly_rate")]
        public decimal? HourlyRate { get; set; }

        [Column("is_verified")]
        public bool IsVerified { get; set; } = false;

        [Column("verification_status")]
        public string VerificationStatus { get; set; } = "pending";

        [Column("average_rating")]
        public decimal AverageRating { get; set; } = 0.00m;

        [Column("total_reviews")]
        public int TotalReviews { get; set; } = 0;

        [Column("verified_at", TypeName = "timestamp without time zone")]
        public DateTime? VerifiedAt { get; set; }

        [Column("created_at", TypeName = "timestamp without time zone")]
        public DateTime CreatedAt { get; set; } = DateTime.SpecifyKind(DateTime.Now, DateTimeKind.Unspecified);

        // Nowe pola do zarządzania zawieszeniem konta specjalisty 
        [Column("is_suspended")]
        public bool IsSuspended { get; set; } = false;

        [Column("suspended_at", TypeName = "timestamp without time zone")]
        public DateTime? SuspendedAt { get; set; }



        public virtual User User { get; set; } = null!;
        public virtual ICollection<SpecialistService> Services { get; set; } = new List<SpecialistService>();
        public virtual ICollection<ServiceArea> ServiceAreas { get; set; } = new List<ServiceArea>();
        public virtual ICollection<SpecialistAvailability> Availabilities { get; set; } = new List<SpecialistAvailability>();
        public virtual ICollection<BookedSlot> BookedSlots { get; set; } = new List<BookedSlot>();
        public virtual ICollection<Appointment> Appointments { get; set; } = new List<Appointment>();
        public virtual ICollection<Review> Reviews { get; set; } = new List<Review>();
        public virtual ICollection<SpecialistQualification> Qualifications { get; set; } = new List<SpecialistQualification>();
        public virtual ICollection<VerificationLog> VerificationLogs { get; set; } = new List<VerificationLog>();
    }
}